### PR TITLE
Overhaul of status API client

### PR DESCRIPTION
**Pull request by khellang**
Added the new endpoint for [combined commit statuses](https://developer.github.com/v3/repos/statuses/#get-the-combined-status-for-a-specific-ref), replaced a legacy URL and added some XML docs :lipstick: 

Tell me if you want the commits squashed after a review :smile:

### CODE REVIEW

**Comment by khellang** [comment]
Rebased and made setters `protected` to address #650 :smile:

**Comment by shiftkey** [comment]
@khellang :thumbsup: added to my review queue

File: .gitignore [hunk 1]
```diff
@@ -72,6 +72,7 @@ tools/FAKE.Core
 tools/SourceLink.Fake
 tools/xunit.runners
 *.ncrunch*
+*.GhostDoc.xml
```
Comment (Haacked) [inline .gitignore]: Hmmm, I might want to start using GhostDoc again. It helps with the boilerplate ctor descriptions etc. What's in the .xml doc? Would we want to include this in the repo if we start using GhostDoc?

Comment (khellang) [inline .gitignore]: It's a really nice tool when having to do a lot of XML docs :smile: The file is for custom settings and ignore words etc. for spell checking. GhostDoc generates it automatically for each solution I open, so I figured I might as well ignore it so I don't accidentally commit it. If you want it in the repo, I'll drop the commit :smile:

Here's a default file:

```xml
<GhostDoc>
  <SpellChecker>
    <IncludeExtensions>
    </IncludeExtensions>
    <IgnoreExtensions>
    </IgnoreExtensions>
    <IgnoreFiles>
    </IgnoreFiles>
  </SpellChecker>
<HelpConfigurations selected="HelpFile">
  <HelpConfiguration name="HelpFile">
    <OutputPath>.\Help</OutputPath>
    <HelpFileName>Bankeren</HelpFileName>
    <ImageFolderPath />
    <HtmlFormats>
      <HtmlHelp>true</HtmlHelp>
      <MSHelpViewer>false</MSHelpViewer>
      <MSHelp2>false</MSHelp2>
      <Website>false</Website>
    </HtmlFormats>
    <IncludeScopes>
      <Public>true</Public>
      <Internal>false</Internal>
      <Protected>false</Protected>
      <Private>false</Private>
      <Inherited>true</Inherited>
      <InheritedFromReferences>true</InheritedFromReferences>
      <EnableTags>false</EnableTags>
      <TagList />
    </IncludeScopes>
    <ResolveCrefLinks>true</ResolveCrefLinks>
    <HeaderText />
    <FooterText />
    <SelectedProjects />
  </HelpConfiguration>
</HelpConfigurations>
</GhostDoc>
```

Comment (Haacked) [inline .gitignore]: @shiftkey what do you think?

Comment (shiftkey) [inline .gitignore]: Someone would need to convince me of the value of GhostDoc -  I have nothing but bad memories from the time I last touched it.

:thumbsup: to ignoring for now

File: Octokit.Tests.Integration/Clients/CommitStatusClientTests.cs [hunk 2]
```diff
@@ -28,6 +28,27 @@ public class TheGetAllMethod
         }
     }
 
+    public class TheGetCombinedMethod
+    {
+        [IntegrationTest]
+        public async Task CanRetrieveCombinedStatus()
```
Comment (shiftkey) [inline Octokit.Tests.Integration/Clients/CommitStatusClientTests.cs]: Add in a test for this commit - it has multiple statuses, so you can test that we're handling collections correctly: https://github.com/libgit2/libgit2sharp/commit/f54529997b6ad841be524654d9e9074ab8e7d41d

File: Octokit.sln.DotSettings [hunk 3]
```diff
@@ -262,5 +262,7 @@ II.2.12 &lt;HandlesEvent /&gt;&#xD;
 	<s:String x:Key="/Default/CodeStyle/Naming/CSharpNaming/PredefinedNamingRules/=PrivateInstanceFields/@EntryIndexedValue">&lt;Policy Inspect="True" Prefix="_" Suffix="" Style="aaBb" /&gt;</s:String>
 	<s:String x:Key="/Default/CodeStyle/Naming/CSharpNaming/PredefinedNamingRules/=PrivateStaticFields/@EntryIndexedValue">&lt;Policy Inspect="True" Prefix="_" Suffix="" Style="aaBb" /&gt;</s:String>
 	<s:String x:Key="/Default/CodeStyle/Naming/CSharpNaming/PredefinedNamingRules/=PrivateStaticReadonly/@EntryIndexedValue">&lt;Policy Inspect="True" Prefix="_" Suffix="" Style="aaBb" /&gt;</s:String>
+	<s:Boolean x:Key="/Default/Environment/SettingsMigration/IsMigratorApplied/=JetBrains_002EReSharper_002EPsi_002ECSharp_002ECodeStyle_002ECSharpFileLayoutPatternsUpgrade/@EntryIndexedValue">True</s:Boolean>
```
Comment (shiftkey) [inline Octokit.sln.DotSettings]: R# 9 update?

File: Octokit/Models/Response/CombinedCommitStatus.cs [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+        /// <summary>
+        /// The total number of statuses.
+        /// </summary>
+        public int TotalCount { get; protected set; }
+
+        /// <summary>
+        /// The statuses.
+        /// </summary>
+        // TODO: This ought to be an IReadOnlyList<ApiErrorDetail> but we need to add support to SimpleJson for that.
```
Comment (shiftkey) [inline Octokit/Models/Response/CombinedCommitStatus.cs]: `IReadOnlyList<CommitStatus>` here

File: Octokit.sln.DotSettings (hunk 3 above)
Comment (khellang) [inline Octokit.sln.DotSettings]: No idea really. The lines were added automatically when opening the sln with R# 9. Want me to remove it?

File: Octokit.Tests.Integration/Clients/CommitStatusClientTests.cs (hunk 2 above)
Comment (khellang) [inline Octokit.Tests.Integration/Clients/CommitStatusClientTests.cs]: Just changed the existing test. It didn't really test what we wanted (more or less just copied from the test above :stuck_out_tongue:).

File: Octokit.sln.DotSettings (hunk 3 above)
Comment (shiftkey) [inline Octokit.sln.DotSettings]: Until I get around to codifying a R# settings file for it, let's :fire: these automagic changes...

File: Octokit.Tests.Integration/Clients/CommitStatusClientTests.cs (hunk 2 above)
Comment (shiftkey) [inline Octokit.Tests.Integration/Clients/CommitStatusClientTests.cs]: :ok_hand:

**Comment by khellang** [comment]
:fire: DotSettings - done :ok_hand:

**Pull request merged**